py/bitbox02/bitbox02/communication/bitbox_api_protocol.py

```diff
@@ -407,3 +418,44 @@ def get_info(self) -> Tuple[str, Platform, Union[BitBox02Edition, BitBoxBaseEdit
         unlocked_byte = response[0]
         unlocked = {0x00: False, 0x01: True}[unlocked_byte]
         return (version_str, platform, edition, unlocked)
+
+    def firmware_outdated(self, required_version: semver.VersionInfo) -> None:
+        raise VersionOutdatedException(
+            "The BitBox02's firmware is no up to date. Device: {}, Required: {}".format(
+                self.version, required_version
+            )
+        )
+
+    def library_outdated(self) -> None:
+        raise VersionOutdatedException(
+            "The BitBox02's firmware version {} is too new for this app. Update the app".format(
+                self.version
+            )
+        )
+
+    def check_version(self) -> None:
+        """
+        Returns false if the device has an older firmware version than required
+        and the minimum required version.
+        """
+        _, platform, edition, _ = self.get_info()
```

Comment:
same, set the edition in `__init__` based on the usb descriptor (you can ignore `platform`/base for now, as that needs to come from get_info indeed, but this would need a bigger refactor).

Reply:
Ok, I removed the check for the BitBoxBase entirely now and use the product string from the device dict to get the edition information.